index.bs

```diff
@@ -1610,7 +1886,243 @@ level APIs. The list above includes interface classes for which such high level
+        1. <a>Append</a> |alternate| to <a>this</a>.{{USBInterface/[[alternates]]}}.
+</dl>
+
+### Attributes ### {#usbinterface-attributes}
+
+<dl dfn-type=attribute dfn-for="USBInterface">
+    : <dfn>interfaceNumber</dfn>
+    ::
+        Each interface provides a collection of {{USBInterface/alternates}} identified
+        by a single <code>bInterfaceNumber</code> field found in their <a>interface
+        descriptors</a>. The {{USBInterface/interfaceNumber}} attribute MUST match this
+        field.
+
+        The {{USBInterface/interfaceNumber}} getter steps are:.
+          1. Return <a>this</a>.{{USBInterface/[[interfaceNumber]]}}.
+    
+    : <dfn>alternate</dfn>
+    ::
+        The {{USBInterface/alternate}} attribute SHALL be set to the
+        {{USBAlternateInterface}} that is currently selected for this interface, which
+        by default SHALL be the one with <code>bAlternateSetting</code> equal to
+        <code>0</code>.
+
+        The {{USBInterface/alternate}} getter steps are:
+            1. Let |configuration| be <a>this</a>.{{USBInterface/[[configuration]]}}.
+            1. Let |device| be |configuration|.{{USBConfiguration/[[device]]}}.
+            1. Let |alternateIndex| be 0.
+            1. If <a>this</a>.{{USBInterface/claimed}} is <code>true</code>:
+                1. Let |interfaceIndex| be the result of <a>finding the interface index</a>
+                    with <a>this</a>.{{USBInterface/[[interfaceNumber]]}} and |configuration|.
+                1. If |interfaceIndex| is not equal to <code>-1</code>:
+                    1. Let |interface| be |configuration|.{{USBConfiguration/[[interfaces]]}}[|interfaceIndex|].
+                    1. Set |alternateIndex| to the result of <a>finding the alternate index</a>
+                        with |device|.{{USBDevice/[[selectedAlternateSetting]]}}[|interfaceIndex|] and |interface|.
+            1. <a>Assert</a>: |alternateIndex| is not equal to <code>-1</code>.
+            1. Return <a>this</a>.{{USBInterface/alternates}}[|alternateIndex|].
+        
+        Note: There is at least one alternate setting for an interface according to
+        <a>Interface Descriptor</a> [[!USB31]], as there must at least one alternate setting 
+        in the <a>Interface Descriptor</a>.
+
+    : <dfn>alternates</dfn>
+    ::
+        The {{alternates}} method provides a collection of {{USBAlternateInterface}}
+        objects identified by a single <code>bInterfaceNumber</code> field found in
+        their <a>interface descriptors</a>.
+
+        The {{alternates}} getter steps are:
+          1. Return <a>this</a>.{{USBInterface/[[alternates]]}}.
+
+    : <dfn>claimed</dfn>
+    ::
+        The {{USBInterface/claimed}} attribute SHALL be set to <code>true</code> when
+        the interface is claimed by the current execution context and SHALL be set to
+        <code>false</code> otherwise.
+
+        The {{USBInterface/claimed}} getter steps are:
```

Comment:
As with the `configuration` attribute, define an algorithm for "is interface claimed" and use it here and anywhere that is currently accessing this attribute directly.

Reply:
Sure, added a new algorithm `finding if the interface is claimed` and have it be used in a few places that look into interface claimed state. Updated in https://github.com/WICG/webusb/pull/212/commits/e1b37d2a2893a75f3823dc2e2fe7635c22c1bfc4.